docs/user_guides/single_table/predefined_constraints.rst

```diff
@@ -0,0 +1,410 @@
+
+    employees = load_tabular_demo()
+    employees
+
+
+This step loaded a simple table that gives us some basic details about
+simulated employees from several companies.
+
+If we observe the data closely we will find a few **constraints**:
+
+1. Each ``company`` has employees from two or more ``departments``, but
+   ``department`` names are different across ``companies``. This implies
+   that a ``company`` should only be paired with its own ``departments``
+   and never with the ``departments`` of other ``companies``.
+2. We have an ``age`` column that represents the age of the employee at
+   the date when the data was created and an ``age_when_joined`` that
+   represents the age of the employee when they joined the ``company``.
+   Since all of them joined the ``company`` before the data was created,
+   the ``age_when_joined`` will always be equal or lower than the
+   ``age`` column.
+3. We have a ``years_in_the_company`` column that indicates how many
+   years passed since they joined the company, which means that the
+   ``years_in_the_company`` will always be equal to the ``age`` minus
+   the ``age_when_joined``.
+4. We have a ``salary`` column that should always be rounded to 2
+   decimal points.
+5. The ``age`` column is bounded, since realistically an employee can only be
+   so old (or so young).
+6. The ``full_time``, ``part_time`` and ``contractor`` columns
+   are related in such a way that one of them will always be one and the others
+   zero, since the employee must be part of one of the three categories.
+
+How does SDV Handle Constraints?
+--------------------------------
+
+**SDV** handles constraints using two different strategies:
+
+Transform Strategy
+~~~~~~~~~~~~~~~~~~
+
+When using this strategy, **SDV** applies a transformation to the data
+before learning it in a way that allows the model to better capture the
+data properties. For example, if we have one column that needs to be
+always greater than the other one, SDV can do the following:
+
+1. Replace the higher column with the difference between the two
+   columns, which will always be positive.
+2. Model the transformed data and sample new values.
+3. Recompute the value of the high column by adding the values of the
+   lower column to it.
+
+The **Transform** strategy is very efficient and does not affect the
+speed of the modeling and sampling process, but in some cases might
+affect the quality of the learning process or simply not be possible.
+
+Reject Sampling Strategy
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the cases where applying a **Transform** strategy is not possible or
+may affect the quality of the learning process, **SDV** can apply a
+**Reject Sampling** strategy.
+
+When using this strategy, **SDV** validates the sampled rows, discards
+the ones that do not adjust to the constraint, and re-samples them. This
+process is repeated until enough rows have been sampled.
+
+Predefined Constraints
+-------------------------------
```

Comment:
Suggestion: there is another documentation page for "CustomConstraints" coming where it's on the same level as "Predefined Constraints". Therefore, I would make this a higher header
```rst
Predefined Constraints
======================
```
and I would actually split this into two pages, such that it will be:
```
├── single_table/
    ├── constraints.rst
    ├── predefined_constraints.rst
    └── custom_constraints.rst
```

Otherwise just truncate the number of '--' to be of equal length to the header.